frontend/src/app/pages/exercises/exercise/shared/exercise-map/shared/restricted-zone-popup/restricted-zone-popup.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, OnInit } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { map, Observable, combineLatest } from 'rxjs';
+import type {
+    RestrictedZone,
+    UUID,
+    VehicleTemplate,
+    Vehicle,
+    VehicleRestrictionType,
+} from 'digital-fuesim-manv-shared';
+import type { AppState } from 'src/app/state/app.state';
+import {
+    createSelectRestrictedZone,
+    selectVehicleTemplates,
+    selectVehicles,
+} from 'src/app/state/application/selectors/exercise.selectors';
+import { selectCurrentRole } from 'src/app/state/application/selectors/shared.selectors';
+import { ExerciseService } from 'src/app/core/exercise.service';
+import { PopupService } from '../../utility/popup.service';
+
+type NavIds = 'settings' | 'vehicleRestrictions';
+
+/**
+ * We want to remember the last selected nav item, so the user doesn't have to manually select it again.
+ */
+let activeNavId: NavIds = 'settings';
+
+@Component({
+    selector: 'app-restricted-zone-popup',
+    templateUrl: './restricted-zone-popup.component.html',
+    styleUrls: ['./restricted-zone-popup.component.scss'],
+    standalone: false,
+})
+export class RestrictedZonePopupComponent implements OnInit {
+    // These properties are only set after OnInit
+    public restrictedZoneId!: UUID;
+
+    public restrictedZone$?: Observable<RestrictedZone>;
+    public readonly currentRole$ = this.store.select(selectCurrentRole);
+    public availableVehicleTypes$?: Observable<string[]>;
+
+    public get activeNavId() {
+        return activeNavId;
+    }
+    public set activeNavId(value: NavIds) {
+        activeNavId = value;
+    }
+
+    constructor(
+        private readonly store: Store<AppState>,
+        private readonly exerciseService: ExerciseService,
+        private readonly popupService: PopupService
+    ) {}
+
+    ngOnInit() {
+        this.restrictedZone$ = this.store.select(
+            createSelectRestrictedZone(this.restrictedZoneId)
+        );
+
+        // Alle verfügbaren Fahrzeugtypen sammeln
```

Comment:
German comments (also in the next lines) that (in my opinion) don't provide additional information/context compared to what is written in the code anyways